### ISSUE TITLE
warning: missing documentation for an associated function

### ISSUE THREAD

**Issue by link2xt** [Voultapher/self_cell#42]
`async_imap` crate maintained at https://github.com/async-email/async-imap/ has `#![warn(missing_docs)]` enabled. When `self_cell` is used as in https://github.com/async-email/async-imap/pull/86, CI fails with warnings like
```
warning: missing documentation for an associated function
  --> src/types/name.rs:7:1
   |
7  | / self_cell!(
8  | |     /// A name that matches a `LIST` or `LSUB` command.
9  | |     pub struct Name {
10 | |         owner: Box<ResponseData>,
...  |
16 | |     impl { Debug }
17 | | );
   | |_^
   |
note: the lint level is defined here
  --> src/lib.rs:76:9
   |
76 | #![warn(missing_docs)]
   |         ^^^^^^^^^^^^
   = note: this warning originates in the macro `self_cell` (in Nightly builds, run with -Z macro-backtrace for more info)
```

A solution could be to add docs to generated methods or at least `#[allow(missing_docs)]` attributes.

**Comment by Voultapher** [comment]
Thanks for reporting the issue, I left some comments in the PR.

**Comment by Voultapher** [comment]
I published version 1.0.1 with the fix.
